server/adaptors/integrations/integrations_builder.ts

```diff
@@ -207,7 +207,7 @@ export class IntegrationInstanceBuilder {
     dataSourceMDSId?: string,
     dataSourceMDSLabel?: string
   ): SavedObject[] {
-    if (!dataSource) {
+    if (!dataSourceMDSId) {
```

Comment:
what's the difference? and is `dataSource` still used?

Reply:
Datasource is the flint one. We need to fix this up soon in whole code-base.